src/background/background.ts

```diff
@@ -11,38 +11,53 @@ chrome.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
     if (message.command === "startRecordingBytesTransferred") {
         IBytesRepository.instance.clearBytesTransferred();
 
-        chrome.debugger.attach({ tabId: tabId }, "1.2", () => {
-            chrome.debugger.sendCommand(
-                { tabId: tabId },
-                "Network.enable",
-                {},
-                () => {
-                    if (chrome.runtime.lastError) {
-                        console.error(chrome.runtime.lastError);
-                    }
-                }
-            );
+        chrome.tabs.get(tabId).then((tab) => {
+            if (tab.url && tab.url.startsWith("chrome://")) {
+                sendResponse({
+                    success: false,
+                    message:
+                        "Cannot calculate emissions for a chrome:// URL, e.g. manage extensions page. Please try again on a valid webpage.",
+                });
+            } else {
+                chrome.debugger.attach({ tabId: tabId }, "1.2").then(() => {
+                    chrome.debugger.sendCommand(
+                        { tabId: tabId },
+                        "Network.enable",
+                        {},
+                        () => {
+                            if (chrome.runtime.lastError) {
+                                console.error(chrome.runtime.lastError);
+                            }
+                            sendResponse({
+                                success: true,
+                                message:
+                                    "Successfully started recording bytes transferred.",
+                            });
+                        }
+                    );
+                });
+            }
         });
-
-        sendResponse(true);
     }
 
     if (message.command === "stopRecordingBytesTransferred") {
-        try {
-            await chrome.debugger.detach({ tabId: tabId });
-        } catch (e: unknown) {
-            if (
-                (e as Error).message ===
-                `Debugger is not attached to the tab with id: ${tabId}.`
-            ) {
-                console.warn(
-                    `Tried to detach debugger from tab (tabId: ${tabId}) when there was none attached. `
-                );
-                return;
-            }
-            throw e;
-        }
-        sendResponse(true);
+        chrome.debugger
+            .detach({ tabId: tabId })
```

Comment:
can just do `{tabId}` it will infer the value from the key name if they are the same